config/muxcable.py

```diff
@@ -631,3 +631,242 @@ def setswitchmode(state, port):
+        return True
+    else:
+        click.echo("firmware download failure {}".format(port))
+        return False
+
+
+def perform_activate_firmware(physical_port):
+    import sonic_y_cable.y_cable
+    result = sonic_y_cable.y_cable.activate_firmware(physical_port)
+    if result == sonic_y_cable.y_cable.FIRMWARE_ACTIVATE_SUCCESS:
+        click.echo("firmware activate successful for {}".format(port))
+        return True
+    else:
+        click.echo("firmware activate failure for {}".format(port))
+        return False
+
+
+def perform_rollback_firmware(physical_port):
+    import sonic_y_cable.y_cable
+    result = sonic_y_cable.y_cable.rollback_firmware(physical_port)
+    if result == sonic_y_cable.y_cable.FIRMWARE_ROLLBACK_SUCCESS:
+        click.echo("firmware rollback successful {}".format(port))
+        return True
+    else:
+        click.echo("firmware rollback failure {}".format(port))
+        return False
+
+
+@muxcable.group(cls=clicommon.AbbreviationGroup)
+def firmware():
+    """Configure muxcable firmware command"""
+    pass
+
+
+@firmware.command()
+@click.argument('fwfile', metavar='<firmware_file>', required=True)
+@click.argument('port', metavar='<port_name>', required=True, default=None)
+def download(fwfile, port):
+    """Config muxcable firmware download"""
+
+    per_npu_statedb = {}
+    y_cable_asic_table_keys = {}
+    port_table_keys = {}
+
+    get_per_npu_statedb(per_npu_statedb, port_table_keys)
+
+    if port is not None and port != "all":
+
+        physical_port_list = []
+        get_physical_port_list(port, physical_port_list)
+        physical_port = physical_port_list[0]
+        if per_npu_statedb[asic_index] is not None:
+            y_cable_asic_table_keys = port_table_keys[asic_index]
+            logical_key = "MUX_CABLE_TABLE|{}".format(port)
+            if logical_key in y_cable_asic_table_keys:
+                perform_download_firmware(physical_port, fwfile)
+
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+                sys.exit(CONFIG_FAIL)
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+            sys.exit(CONFIG_FAIL)
+
+    elif port == "all" and port is not None:
+
+        rc = True
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for key in port_table_keys[asic_id]:
+                port = key.split("|")[1]
+
+                physical_port_list = []
+                get_physical_port_list(port, physical_port_list)
+
+                physical_port = physical_port_list[0]
+
+                status = perform_download_firmware(physical_port, fwfile)
+
+                if status is not True:
+                    rc = False
+
+        if rc:
+            sys.exit(CONFIG_SUCCESSFUL)
+        else:
+            sys.exit(CONFIG_FAIL)
```

Comment:
`rc` seems redundant as a bool. Suggest setting `rc = CONFIG_SUCCESSFUL` on line 750, `rc = CONFIG_FAIL` on line 764, and here simply call `sys.exit(rc)`.

Suggest making this change throughout the file.

Reply:
fixed